src/test/java/uk/gov/cshr/service/EmailUpdateServiceTest.java

```diff
@@ -86,70 +94,56 @@ public void setUp() {
-        Optional<EmailUpdate> optionalEmailUpdate = Optional.of(emailUpdate);
-        when(emailUpdateRepository.findByIdentityAndCode(any(Identity.class), anyString())).thenReturn(optionalEmailUpdate);
+        emailUpdate.setEmail(NEW_EMAIL_ADDRESS);
 
-        doNothing().when(identityService).updateEmailAddressAndEmailRecentlyUpdatedFlagToTrue(eq(IDENTITY), eq(emailUpdate.getEmail()));
+        when(identityService.getDomainFromEmailAddress(NEW_EMAIL_ADDRESS)).thenReturn(NEW_DOMAIN);
+
+        doNothing().when(identityService).updateEmailAddressWhereNewEmailIsNotAgency(eq(IDENTITY), eq(emailUpdate.getEmail()));
+        doNothing().when(identityService).updateSpringWithRecentlyEmailUpdatedFlag(any(HttpServletRequest.class), eq(true));
         doNothing().when(emailUpdateRepository).delete(any(EmailUpdate.class));
 
-        // when
-        classUnderTest.updateEmailAddress(request, IDENTITY, "CO");
+        emailUpdateService.updateEmailAddress(request, IDENTITY, emailUpdate);
 
-        // then
-        verify(emailUpdateRepository, times(1)).findByIdentityAndCode(eq(IDENTITY), eq("CO"));
-        verify(identityService, times(1)).updateEmailAddressAndEmailRecentlyUpdatedFlagToTrue(eq(IDENTITY), eq(emailUpdate.getEmail()));
+        verify(identityService, times(1)).updateEmailAddressWhereNewEmailIsNotAgency(identityArgumentCaptor.capture(), eq(emailUpdate.getEmail()));
         verify(identityService, times(1)).updateSpringWithRecentlyEmailUpdatedFlag(any(HttpServletRequest.class), eq(true));
         verify(emailUpdateRepository, times(1)).delete(emailUpdateArgumentCaptor.capture());
 
         EmailUpdate actualDeletedEmailUpdate = emailUpdateArgumentCaptor.getValue();
         assertThat(actualDeletedEmailUpdate.getId(), equalTo(100l));
-    }
-
-    @Test(expected = InvalidCodeException.class)
-    public void givenANonExistentEmailUpdate_whenUpdateEmailAddress_shouldThrowInvalidCodeException(){
-        // given
-        Optional<EmailUpdate> optionalEmailUpdate = Optional.empty();
-        when(emailUpdateRepository.findByIdentityAndCode(any(Identity.class), anyString())).thenReturn(optionalEmailUpdate);
-
-        // when
-        classUnderTest.updateEmailAddress(request, IDENTITY, "CO");
 
-        // then
-        verify(emailUpdateRepository, times(1)).findByIdentityAndCode(eq(IDENTITY), eq("CO"));
-        verify(identityService, never()).updateEmailAddressAndEmailRecentlyUpdatedFlagToTrue(any(Identity.class), anyString());
-        verify(identityService, never()).updateSpringWithRecentlyEmailUpdatedFlag(any(HttpServletRequest.class), anyBoolean());
-        verify(emailUpdateRepository, never()).delete(any(EmailUpdate.class));
+        Identity identity = identityArgumentCaptor.getValue();
+        assertThat(identity.getUid(), equalTo(UID));
+        assertThat(identity.getAgencyTokenUid(), equalTo(null));
     }
 
+    // TODO: 21/05/2020 Test throwing the exception  
```

Comment:
Write tests for throwing exception scenario and delete TODO.

Reply:
Good catch. It was originally a stale todo comment, have since moved exception handling to IdentityService.java. Have not created a new test for this scenario:

`identityRepositoryShouldThrowExceptionIfNotFound()`